packages/core/src/client.ts

```diff
@@ -316,6 +316,103 @@ export class Client extends AsyncEventEmitter<MappedEvents> {
 		return { members, nonce, notFound, presences };
 	}
 
+	/**
+	 * Requests soundboard sounds from the gateway and returns an async iterator that yields the data from each soundboard sounds event.
+	 *
+	 * @see {@link https://discord.com/developers/docs/topics/gateway-events#request-soundboard-sounds}
+	 * @param options - The options for the request
+	 * @param timeout - The timeout for waiting for each soundboard sounds
+	 * @example
+	 * Requesting soundboard sounds for specific guilds
+	 * ```ts
+	 * for await (const { soundboardSounds } of this.requestSoundboardSoundsIterator({
+	 *	guild_ids: ['1234567890', '9876543210'],
+	 * 	})) {
+	 *	console.log(soundboardSounds);
+	 *}
+	 * ```
+	 */
+	public async *requestSoundboardSoundsIterator(options: GatewayRequestSoundboardSoundsData, timeout = 10_000) {
+		const shardCount = await this.gateway.getShardCount();
+		const shardIds = groupBy(options.guild_ids, (guildId) => calculateShardId(guildId, shardCount));
+
+		const controller = new AbortController();
+
+		let timer: NodeJS.Timeout | undefined = createTimer(controller, timeout);
+
+		for (const [shardId, guildIds] of shardIds) {
+			await this.gateway.send(shardId, {
+				op: GatewayOpcodes.RequestSoundboardSounds,
+				// eslint-disable-next-line id-length
+				d: {
+					...options,
+					guild_ids: guildIds,
+				},
+			});
+		}
+
+		try {
+			const iterator = AsyncEventEmitter.on(this, GatewayDispatchEvents.SoundboardSounds, {
+				signal: controller.signal,
+			});
+
+			const guildIds = new Set(options.guild_ids);
+
+			for await (const [{ data }] of iterator) {
+				clearTimeout(timer);
+				timer = undefined;
+
+				yield {
+					guildId: data.guild_id,
+					soundboardSounds: data.soundboard_sounds,
+				};
+
+				guildIds.delete(data.guild_id);
```

Comment:
Shouldn't we yield only if the guild id is in the list of guild ids the user provided?

Reply:
When would the guild id not be in the list of guild ids the user provided though?

Reply:
Multiple requests?